File: CSharp/DP.Domain/Samples/Prototype/PrototypeStore.cs
using System.Collections.Generic;

namespace DP.Domain.Samples.Prototype
{
    /// <summary>
    /// 存放Prototype objects的Store
    /// </summary>
    public class PrototypeStore
    {
        private Dictionary<StoreEnum, IPrototype> prototypes = null;

        public PrototypeStore()
        {
            prototypes = new Dictionary<StoreEnum, IPrototype>();
        }

        /// 加入新的Prototype
        public void Add(StoreEnum store, IPrototype prototype)
        {
            prototypes.Add(store, prototype);
        }
        /// 由ID取得特定Prototype
        public IPrototype Get(StoreEnum store)
        {
            var prototype = (IPrototype)this.prototypes[store].Clone();
            return prototype;
        }
    }

    public enum StoreEnum
    {
        Fatbook,
        Goople,
        Amozoo
    }
}

File: Python/Samples/Prototype/PrototypeStore.py
from enum import Enum
from Prototypes import Prototype, PrototypeFatbook, PrototypeGoople

class StoreEnum(Enum):
    Goople = 1,
    Fatbook = 2,
    Amozoo = 3


class PrototypeStore:

    _prototypes = {}

    def add(self, store=StoreEnum, prototype=Prototype):
        self._prototypes[store] = prototype

    def get(self, store=StoreEnum):
        return self._prototypes[store].clone()